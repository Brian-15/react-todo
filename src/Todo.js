const Todo = ({ id, text, removeTask }) => {
    return (
        <li id={ id } key={id} data-testid='task'>
            <p>{ text }</p>
            <button onClick={ removeTask } data-testid='remove-btn'>X</button>
        </li>
    )
};

export default Todo;