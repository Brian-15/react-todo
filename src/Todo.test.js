import { render } from '@testing-library/react';
import Todo from './Todo.js';

describe('Todo component', () => {
    test('smoke test: renders component', () => {
        render(<Todo text={'test text'} />);
    });

    test('snapshot test', () => {
        const { asFragment } = render(<Todo text={'test text'} />);
        expect(asFragment()).toMatchSnapshot();
    });
});