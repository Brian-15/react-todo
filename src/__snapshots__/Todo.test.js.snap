// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component snapshot test 1`] = `
<DocumentFragment>
  <li
    data-testid="task"
  >
    <p>
      test text
    </p>
    <button
      data-testid="remove-btn"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
